xclim/sdba/measures.py

```diff
@@ -442,3 +442,69 @@ def _scorr(
 scorr = StatisticalPropertyMeasure(
     identifier="Scorr", aspect="spatial", compute=_scorr, allowed_groups=["group"]
 )
+
+
+def _taylordiagram(
+    sim: xr.DataArray,
+    ref: xr.DataArray,
+    dim: str = "time",
+    group: str | Grouper = "time",
+) -> xr.DataArray:
+    """Taylor diagram.
+
+    Compute the respective standard deviations of a simulation and a reference array, as well as the Pearson
+    correlation coefficient between both, all necessary parameters to plot points on a Taylor diagram.
+
+    Parameters
+    ----------
+    sim : xr.DataArray
+        data from the simulation (a time-series for each grid-point)
+    ref : xr.DataArray
+        data from the reference (observations) (a time-series for each grid-point)
+    dim : str
+        Dimension across which the correlation and standard deviation should be computed.
+    group : str
+        Compute the property and measure for each temporal groups individually.
+        Currently not implemented.
+
+
+    Returns
+    -------
+    xr.DataArray, [measures]
+        Standard deviations of sim, ref and correlation coefficient between both.
+    """
+    corr = xr.corr(sim, ref, dim=dim)
+
+    ref_std = ref.std(dim=dim, skipna=True, keep_attrs=True)
+    sim_std = sim.std(dim=dim, skipna=True, keep_attrs=True)
+
+    coords = {"taylor_param": ["ref_std", "sim_std", "corr"]}
+    dims = ["taylor_param"]
+
+    for coord in ref_std.coords:
+        coords[coord] = ref_std[coord]
+    for dim in ref_std.dims:
+        dims.append(dim)
+
+    out = xr.DataArray(
+        [ref_std, sim_std, corr],
+        coords=coords,
+        dims=dims,
+        attrs={"correlation_type": "Pearson correlation coefficient"},
+    )
+
+    if hasattr(ref_std, "units") and hasattr(sim_std, "units"):
+        if ref_std.units == sim_std.units:
+            out.attrs["units"] = ref_std.units
+        else:
+            raise ValueError("Simulation and reference units must be identical.")
```

Comment:
Go see the definition of `StatisticalPropertyMeasure` above. It automatically convert ref units to sim units, so this is never going to happen.